M2/Macaulay2/packages/ReesAlgebra.m2

```diff
@@ -209,26 +207,50 @@ isLinearType(Module, RingElement):= o-> (N,a)->(
      J := ideal((vars S) * P);
      ((gens I) % J) == 0)
 
-normalCone = method(TypicalValue => Ring, 
-    	    Options => {
+-- normalCone = method(TypicalValue => Ring, 
+--     	    Options => {
+-- 	  DegreeLimit => {},
+-- 	  BasisElementLimit => infinity,
+-- 	  PairLimit => infinity,
+-- 	  MinimalGenerators => true,
+-- 	  Strategy => null,
+-- 	  Variable => "w"
+-- 	  }
+-- )
+-- normalCone(Ideal) := o -> I -> (
+--      RI := reesAlgebra(I,o);
+--      RI/promote(I,RI)
+--      )
+
+-- normalCone(Ideal, RingElement) := o -> (I,a) -> (
+--      RI := reesAlgebra(I,a,o);
+--      RI/promote(I,RI)     
+--      )
+
+normalConeOptions = {
 	  DegreeLimit => {},
 	  BasisElementLimit => infinity,
 	  PairLimit => infinity,
 	  MinimalGenerators => true,
 	  Strategy => null,
 	  Variable => "w"
 	  }
-)
-normalCone(Ideal) := o -> I -> (
+
+
+normalCone Ideal := Ring => normalConeOptions >> o -> I -> (
      RI := reesAlgebra(I,o);
      RI/promote(I,RI)
      )
 
-normalCone(Ideal, RingElement) := o -> (I,a) -> (
+normalCone(Ideal, RingElement) := Ring => normalConeOptions >> o  -> (I,a) -> (
      RI := reesAlgebra(I,a,o);
      RI/promote(I,RI)     
      )
 
+associatedGradedRing = method(Options => normalConeOptions)
```

Comment:
```suggestion
associatedGradedRing = method(TypicalValue => Ring, Options => normalConeOptions)
```

Reply:
We placed the return type into each usage, which we feel is a better place to document the return type

Reply:
Is this a general preference against using `TypicalValue` option of `method`? Can I ask why?